src/sentry/models/outbox.py

```diff
@@ -94,22 +89,274 @@ class OutboxCategory(IntEnum):
+            category=self,
+            object_identifier=object_identifier,
+            payload=payload,
+        )
+
+    def as_control_outboxes(
+        self,
+        region_names: Collection[str],
+        model: Any | None = None,
+        payload: Any | None = None,
+        shard_identifier: int | None = None,
+        object_identifier: int | None = None,
+        outbox: Type[ControlOutboxBase] | None = None,
+    ) -> List[ControlOutboxBase]:
+
+        scope = self.get_scope()
+
+        shard_identifier, object_identifier = self.infer_identifiers(
+            scope, model, object_identifier=object_identifier, shard_identifier=shard_identifier
+        )
+
+        Outbox = outbox or ControlOutbox
+
+        return [
+            Outbox(
+                shard_scope=scope,
+                shard_identifier=shard_identifier,
+                category=self,
+                object_identifier=object_identifier,
+                region_name=region_name,
+                payload=payload,
+            )
+            for region_name in region_names
+        ]
+
+    def infer_identifiers(
+        self,
+        scope: OutboxScope,
+        model: Optional[BaseModel],
+        *,
+        object_identifier: int | None,
+        shard_identifier: int | None,
+    ) -> Tuple[int, int]:
+        from sentry.models.organization import Organization
+        from sentry.models.user import User
+
+        assert (model is not None) ^ (
+            object_identifier is not None
+        ), "Either model or object_identifier must be specified"
+
+        if model is not None and hasattr(model, "id"):
+            object_identifier = model.id
+
+        if shard_identifier is None and model is not None:
+            if scope == OutboxScope.ORGANIZATION_SCOPE:
+                if isinstance(model, Organization):
+                    shard_identifier = model.id
+                elif hasattr(model, "organization_id"):
+                    shard_identifier = model.organization_id
+            if scope == OutboxScope.USER_SCOPE:
+                if isinstance(model, User):
+                    shard_identifier = model.id
+                elif hasattr(model, "user_id"):
+                    shard_identifier = model.user_id
+
+        assert (
+            model is not None
+        ) or shard_identifier is not None, "Either model or shard_identifier must be specified"
+
+        assert object_identifier is not None
+        assert shard_identifier is not None
+        return shard_identifier, object_identifier
+
+
+def scope_categories(enum_value: int, categories: Set[OutboxCategory]) -> int:
+    _outbox_categories_for_scope[enum_value] = categories
+    inter = _used_categories.intersection(categories)
+    assert not inter, f"OutboxCategories {inter} were already registered to a different scope"
+    _used_categories.update(categories)
+    return enum_value
+
+
+class OutboxScope(IntEnum):
+    ORGANIZATION_SCOPE = scope_categories(
```

Comment:
Now, when configuring new OutboxCategory or OutboxScope, you are required to explicitly state the intended heirarchy.  This both self documents and prevents inconsistency in call sites.